autopilot/agent.go

```diff
@@ -24,14 +24,15 @@ type Config struct {
 	// accidentally attempt to open a channel with ourselves.
 	Self *btcec.PublicKey
 
-	// Heuristic is an attachment heuristic which will govern to whom we
-	// open channels to, and also what those channels look like in terms of
-	// desired capacity. The Heuristic will take into account the current
-	// state of the graph, our set of open channels, and the amount of
-	// available funds when determining how channels are to be opened.
-	// Additionally, a heuristic make also factor in extra-graph
-	// information in order to make more pertinent recommendations.
-	Heuristic AttachmentHeuristic
+	// Heuristic is a super-heuristic that will combine scores from its
+	// active sub-heuristics to govern to whom we open channels to, and
+	// also what those channels look like in terms of desired capacity. The
+	// Heuristic will take into account the current state of the graph, our
+	// set of open channels, and the amount of available funds when
+	// determining how channels are to be opened.  Additionally, a
+	// heuristic make also factor in extra-graph information in order to
+	// make more pertinent recommendations.
+	Heuristic SuperHeuristic
```

Comment:
Yes, it was nicer that the interface supported both a regular heuristic and a super heuristic like in the previous iterations.

What about this proposal: 
1. let the `ScoreSettable`'s method `SetNodeScores` take the targetted heuristic name.
2. In `Manager.SetNodeScores`, we assert whether the active heuristic is `ScoreSettable`, if so set the scores, passing along the targeted name of the heuristic.
3. In the `SetNodeScores` method of the heuristic, we choose if we should set the scores (for regular heuristics, and if the name matches) or recursively to sub-heuristics. So it will be up to the regular "leaf" heuristics to decide whether they should apply the scores based on the name passed.
4. They can return a `bool` or specific error that indicates whether a matching leaf-heuristic was found.

cc @cfromknecht

Reply:
SGTM, this sounds like a good compromise :)